tests/test_other.py

```diff
@@ -1305,6 +1305,10 @@ def tmp_source(name, code):
     run_process([PYTHON, EMCC] + args + libs)
     self.assertContained('result: 42', run_js('a2.out.js'))
 
+  # The fastcomp path will deliberatly ignore duplicate input file in order
+  # to allow "libA.so" on the command line twice. The is not realy .so support
+  # and the .so files are reallly bitcode.
```

Comment:
do native linkers not allow linking in the same library twice (with no effects, same as if it were linked once)? or is that just for `-l` but not for providing an `.so` file?

Reply:
They may for .so files, because of the native of .so files.  And for .a files for a different reason.

But we don't support .so files in that sense.   Our support for .so files should be removed IMHO since its confusing and doesn't do what people would want from .so file support.